HW0/P4/P4.py

```diff
@@ -65,12 +91,43 @@
 
     # B = ?
     # C = ?
-
+    B = np.diag([bx, by, bz, bvx, bvy, bvz])
+    C = np.hstack( [np.diag([rx, ry, rz]), np.zeros([3,3])] )
+    
     # Initial conditions for s0 and Sigma0
-    # Compute the rest of sk using Eqs (2), (3), (4), and (5)
+    s0 = np.array([0,0,2,15,3.5,4.0]).reshape([-1,1])
+    sigma0 = 0.01 * np.identity(len(A))
+    
+    # (2)
+    def predictS(sk):
+        return np.dot(A,sk) + a
+    # (3)
+    def predictSig(sigmak):
+        return np.linalg.inv( np.dot(np.dot(A, sigmak), A.T) + np.dot(B, B.T) )
+    # (4)
+    def updateSig(sigmatilt):
+        return np.linalg.inv( sigmatilt + np.dot(C.T, C) )
+    # (5)
+    def updateS(sigmak1, sigmatilt, stilt, mk1):
+        return np.dot( sigmak1, np.dot(sigmatilt, stilt) + np.dot(C.T, mk1) )
+
+    # Compute the rest of sk using Eqs (2), (3), (4), and (5)    
+    s_kal = np.empty([6,K])
+    
+    s_nextkal = s0.copy()
+    sigma_nextkal = sigma0.copy()
+    for i in xrange(K):
+        s_kal[:,i] = s_nextkal.reshape(-1)
+        s_tilt = predictS(s_nextkal)
+        sigma_tilt = predictSig(sigma_nextkal)
+        sigma_nextkal = updateSig(sigma_tilt)
+        mk1 = s_obs[i].reshape([-1,1])
+        s_nextkal = updateS(sigma_nextkal, sigma_tilt, s_tilt, mk1)
+    s_kal = s_kal.T
 
     # ax.plot(x_coords, y_coords, z_coords,
     #         '-r', label='Filtered trajectory')
+    ax.plot(s_kal[:,0], s_kal[:,1], s_kal[:,2], '-r', label='Filtered trajectory')
 
     # Show the plot
     ax.legend()
```

Comment:
I got the same plot. I would suggest adding the plot as a .png just for reference, even though the problem doesn't explicitly ask for it. Otherwise all looks good to me!